### DISCUSSION TITLE
Adding extra attributes to MappedColumn / keep track of columns order

### DISCUSSION THREAD

**Discussion by vinraspa** [sqlalchemy/sqlalchemy#9516] [Usage Questions] [answered]
Is there a smart way to add some extra attributes to a `MappedColumn` instance (obtained directly or through `sqlalchemy.orm.mapped_column()` method?

Explanation: actually, in my project, I must display SQLite data in a Qt's `QTableView`. The first version of my program did not use SQLalchemy but used a quick and dirty home-made ORM which was of course quite limited but in which I could declare SQL columns with types, relationships etc. **plus** GUI-related parameters like text color, alignment, tool-tip text, ...

My first idea is to subclass `MappedColumn` but
1. I fear that in the long run this choice will play tricks on me (I don't have enough of an overview to be sure);
2. In doc, I see: _To construct orm.MappedColumn objects, use the `orm.mapped_column` constructor function._ I guess that there many good reasons to follow those recommendations.

So what is the cleanest and safest way to achieve what I want to do?

For the same purpose, I would like to keep track of the order in which the columns were declared. Any suggestion?

**Comment by zzzeek** [comment, marked as answer]
easy enough use the [.info](https://docs.sqlalchemy.org/en/20/core/metadata.html#sqlalchemy.schema.Column.params.info) parameter, a dictionary with whatever you want in it

**Reply by CaselIT** [reply]
An info parameter is present in most sqlalchemy constructs, like tables, connections, sesstions
